profitbricks-rest/src/main/java/org/apache/jclouds/profitbricks/rest/domain/Lan.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.jclouds.profitbricks.rest.domain;
+
+import com.google.auto.value.AutoValue;
+import org.jclouds.javax.annotation.Nullable;
+import org.jclouds.json.SerializedNames;
+
+@AutoValue
+public abstract class Lan {
+
+   public abstract String id();
+
+   public abstract String type();
+
+   public abstract String href();
+
+   @Nullable
+   public abstract Metadata metadata();
+
+   @Nullable
+   public abstract Properties properties();
+
+   @Nullable
+   public abstract Entities entities();
```

Comment:
The tree of hierarchy, with DataCenter as its root is 'filtered' by the depth level. So I think it will be always there for DataCenter, since it is the root. If however, at some point in future, there would be a grouping above datacenters (perhaps Region?), and if the depth is too low, then it could become null.

It is in a sense relative, if one is requesting an object from the api. Then the object's meta will not be null, but its childrens' meta may be, depending on the depth.

I think it would be sensible to have them nullable as well.